flocker/node/functional/test_docker.py

```diff
@@ -208,6 +208,23 @@ def started(_):
         d.addCallback(started)
         return d
 
+    def test_list_image_data_cached(self):
+        """
+        ``DockerClient.list`` will only an inspect an image ID once, caching
```

Comment:
A different way to demonstrate this; gross but workable:
1. `DockerClient.list()` to get container's info.
2. Patch underlying Docker client so that `inspect_image` returns (for that particular image) different data or maybe fails.
3. Run `DockerClient.list()` again, note it still has correct info.